bundler/lib/bundler/rubygems_ext.rb

```diff
@@ -61,16 +61,97 @@ def open_file_with_lock(path, &block)
   require "rubygems/platform"
 
   class Platform
-    JAVA  = Gem::Platform.new("java")
-    MSWIN = Gem::Platform.new("mswin32")
-    MSWIN64 = Gem::Platform.new("mswin64")
-    MINGW = Gem::Platform.new("x86-mingw32")
-    X64_MINGW = [Gem::Platform.new("x64-mingw32"),
-                 Gem::Platform.new("x64-mingw-ucrt")].freeze
-    UNIVERSAL_MINGW = Gem::Platform.new("universal-mingw")
-    WINDOWS = [MSWIN, MSWIN64, UNIVERSAL_MINGW].flatten.freeze
-    X64_LINUX = Gem::Platform.new("x86_64-linux")
-    X64_LINUX_MUSL = Gem::Platform.new("x86_64-linux-musl")
+    unless respond_to?(:generic)
+      JAVA  = Gem::Platform.new("java")
+      MSWIN = Gem::Platform.new("mswin32")
+      MSWIN64 = Gem::Platform.new("mswin64")
+      MINGW = Gem::Platform.new("x86-mingw32")
+      X64_MINGW = [Gem::Platform.new("x64-mingw32"),
+                   Gem::Platform.new("x64-mingw-ucrt")].freeze
+      UNIVERSAL_MINGW = Gem::Platform.new("universal-mingw")
+      WINDOWS = [MSWIN, MSWIN64, UNIVERSAL_MINGW].flatten(1).freeze
+      X64_LINUX = Gem::Platform.new("x86_64-linux")
+      X64_LINUX_MUSL = Gem::Platform.new("x86_64-linux-musl")
+
+      GENERICS = [RUBY, JAVA, *WINDOWS].freeze
```

Comment:
Is adding `RUBY` to `GENERICS` necessary?

Reply:
not strictly necessary, but it means the GENERICS_CACHE can be pre-primed for all the generic platforms the same way

Reply:
the benefit of putting it in the cache is then you just get a cache hit, vs needing to run the fallback proc

Reply:
(where "ruby" => "ruby" will then get stored)

Reply:
What I'm trying to say is that the fallback proc will never run anyways, because `RUBY` is a `String`, not a `Gem::Platform` and does not have a `#generic` method. `Gem::Platform.generic(platform)` special cases `RUBY` to return itself, and only when the given platform is not `RUBY` it calls `Gem::Platform#generic`. Makes sense? 🤔